Add JAFFE dataset

welch opened the pull request:
I wanted to take this code for a spin, but the RaFD data is pretty locked down without a .edu mailing address.

Here's an interesting dataset of 10 female Japanese faces acting out standard expressions, with FACS scores by a human panel. You get finer-grained scoring with multiple emotions per instance.
http://www.kasrl.org/jaffe_info.html
(noncommercial use, but available to regular Joes like me for download)

Not meant for merge, as something's Not Quite Right and the identity images I get look like this:
![00000](https://cloud.githubusercontent.com/assets/1497207/21860667/7f165622-d7e6-11e6-86f0-1f32ba8bca99.jpg)

Sending this along in case anyone else wants it as a starting place. (also, at three datasets, you're ready for some refactoring).

welch committed:
add use_jaffee flag throughout

welch committed:
add loader for JAFFE dataset

welch committed:
generate inputs for JAFFE model. get some params from model file rather than hard-wiring in code

kaihuchen commented:
I am not affiliated with the academia, but I nonetheless  just went ahead to send the RaFD people a request with some supporting evidence of my identity and my research, and they were nice enough to still grant me access to the dataset.

flynnmd committed:
Set initial shape for JAFFE images

flynnmd committed:
Use num_ids from params dict in GenParser

flynnmd committed:
Initialize identity vector to zero

Junk data getting through with np.empty was causing the identity input to
basically have no effect.

zo7 commented:
A few bugs, but got it to work:

![Random JAFFE Generations](http://i.imgur.com/9t6KfjO.jpg)

But yeah, this project needs a huge refactoring before any more datasets can be added. I'm working on something more general, but that's going to be a separate project I think since this one was only really intended to work with RaFD.
